cas-server-webapp-support/src/main/java/org/jasig/cas/web/flow/AcceptableUsagePolicyFormAction.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.web.flow;
+
+import org.jasig.cas.authentication.Credential;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.binding.message.MessageContext;
+import org.springframework.webflow.action.EventFactorySupport;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Webflow action to receive and record the AUP response.
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public class AcceptableUsagePolicyFormAction {
+
+    /** Event id to signal the policy needs to be accepted. **/
+    protected static final String EVENT_ID_MUST_ACCEPT = "mustAccept";
+
+    /** Logger instance. **/
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final Map<String, Boolean> policyMap = new ConcurrentHashMap<>();
```

Comment:
No, I would think any other impls would want to extend the class itself and handle the storage mechanism. This is for a super simple use case and more like for demonstration purposes. Almost ALL impls would want to extend this class to look into a db or ldap server.